components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -933,15 +934,14 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
     }
 
-    public async resolveContext(ctx: TraceContextWithSpan, contextUrl: string): Promise<WorkspaceContext> {
+    public async resolveContext(
+        ctx: TraceContextWithSpan,
+        contextUrl: string,
+        options?: ParseContextOptions,
+    ): Promise<WorkspaceContext> {
         const user = await this.checkAndBlockUser("resolveContext");
-        const normalizedCtxURL = this.contextParser.normalizeContextURL(contextUrl);
-        try {
-            return await this.contextParser.handle(ctx, user, normalizedCtxURL);
-        } catch (error) {
-            this.handleError(error, { userId: user.id }, normalizedCtxURL);
-            throw error;
-        }
+        const { context } = await this.contextService.parseContext(user, contextUrl, options);
```

Comment:
It's safe to use `ContextService` because what we do in addition is to check if prebuild / snapshot / org is exits and has access to it